src/main/java/org/logstash/dissect/Delimiter.java

```diff
@@ -0,0 +1,42 @@
+package org.logstash.dissect;
+
+import org.logstash.dissect.search.DelimiterSearchStrategy;
+
+public final class Delimiter {
+    private final byte[] needle;
+    private final String delimiter;
+    private final int size;
+    private DelimiterSearchStrategy strategy;
+
+    public Delimiter(String delim) {
+        delimiter = delim;
+        needle = delim.getBytes();
+        size = needle.length;
```

Comment:
Do you need a dedicated field for that? You could just call `needle.length` in `#size()`.